packages/components/src/dynamic-button/dynamic-button.tsx

```diff
@@ -0,0 +1,92 @@
+import { forwardRef } from 'react'
+
+import { ArrowDownIcon, MentionIcon } from '@status-im/icons/12'
+import { Stack, styled } from '@tamagui/core'
+
+import { Shadow } from '../shadow'
+import { Text } from '../text'
+
+import type { GetVariants } from '../types'
+import type { ColorTokens, StackProps } from '@tamagui/core'
+import type { Ref } from 'react'
+import type { PressableProps } from 'react-native'
+
+type Variants = GetVariants<typeof Button>
+
+type Props = PressableProps & {
+  type: Variants['type']
+  count: number
+}
+
+const DynamicButton = (props: Props, ref: Ref<HTMLButtonElement>) => {
+  const { type, count, ...pressableProps } = props
+
+  const color: ColorTokens = '$white-100'
+  const showCount = Boolean(count)
+
+  return (
+    <Shadow variant="$2" borderRadius={999}>
+      <Button
+        {...(pressableProps as unknown as StackProps)} // TODO: Tamagui has incorrect types for PressableProps
```

Comment:
interesting, does this `reset` types to completely different shape?

Reply:
The arguments of `onHoverIn` and `onHoverOut` handlers differ, but functionality is equivalent.

Reply:
And yes, the first `as unknown` type cast enables you to type cast it again to anything you want.